torch/distributed/distributed_c10d.py

```diff
@@ -1400,7 +1425,7 @@ def all_gather_object(object_list, obj, group=group.WORLD):
     # Gather all local sizes. This is so that we can find the max size, and index
     # until the correct size when deserializing the tensors.
     group_size = get_world_size(group=group)
-    object_sizes_tensor = torch.zeros(group_size, dtype=int, device=current_device)
+    object_sizes_tensor = torch.zeros(group_size, dtype=int, device=current_device)  # type: ignore[call-overload]
```

Comment:
If is_nccl_backend is set, current_device is an int, which will not be accepted by torch.zeros().